lib/structs/audit_log_change.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Crux.Structs.AuditLogChange do
+  @moduledoc """
+    Represents a Discord [Audit Log Change Object](https://discordapp.com/developers/docs/resources/audit-log#audit-log-change)
+  """
+  @behaviour Crux.Structs
+
+  alias Crux.Structs
+  alias Crux.Structs.{Overwrite, Role, Util}
+
+  defstruct(
+    new_value: nil,
+    old_value: nil,
+    key: nil
+  )
+
+  @type audit_log_change_value :: String.t() | integer() | boolean() | [Role] | [Overwrite]
```

Comment:
Might be worth mentioning that `Role`s only contain id and name.

---
`Role.t()` and `Overwrite.t()` 👀  

---
`new_value` and `old_value` may also be a Snowflake, no clue how to properly detect and parse those however. ~~Maybe via regex?~~ Would easily behave incorrectly with role names.
Maybe just documentation about them being a string then?

Reply:
I think I have a solution, by checking if keys end with _id